group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -592,6 +607,45 @@ public void validateOffsetFetch(
         validateMemberEpoch(memberEpoch, member.memberEpoch());
     }
 
+    /**
+     * Validates the OffsetDelete request.
+     */
+    @Override
+    public void validateOffsetDelete() throws GroupIdNotFoundException {
+        if (state() == ConsumerGroupState.DEAD) {
+            throw new GroupIdNotFoundException(String.format("Group %s is in dead state.", groupId));
+        }
```

Comment:
I think that a consumer group will actually never transition to Dead. We could actually remove this state.